Neos.ContentGraph.DoctrineDbalAdapter/src/ContentGraphAdapterFactoryBuilder.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Neos\ContentGraph\DoctrineDbalAdapter;
+
+use Neos\ContentRepository\Core\Factory\ProjectionFactoryDependencies;
+use Neos\ContentRepository\Core\Feature\ContentGraphAdapterFactoryBuilderInterface;
+use Neos\ContentRepository\Core\Infrastructure\DbalClientInterface;
+
+/**
+ * Builder to combine injected dependencies and ProjectionFActoryDependencies into a ContentGraphAdapterFactory
+ * @internal
+ */
+class ContentGraphAdapterFactoryBuilder implements ContentGraphAdapterFactoryBuilderInterface
```

Comment:
I wouldn't mind getting around having this, and am happy for suggestions, I don't really see it though. An ugly option would be to to have something along the lines of:

`ContentGraphAdapterFactory::injectProjectionDependencies(ProjectionFactoryDependencies $projectionFactoryDependencies)`

and the `ContentGraphAdapterFactory::__construct` just gets the adapter dependencies injected (eg. DBALClient), but then all methods in the ContentGraphAdapterFactory need to throw a "TooEarly" Exception if the projection dependencies were not injected yet.

Reply:
AFAIS we only need it to pass in `ContentRepositoryId`, `NodeTypeManager` and `PropertyConverter` – Maybe we can make these explicit dependencies somehow

Reply:
..or even make the `NodeFactory` an explicit dependency (via some `NodeFactoryInterface`) but I don't know... Let's discuss